File: archived/2022-09/icpc-d.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

vector<long long> foo(long long n, vector<long long>& A, vector<long long>& B, vector<long long>& C) {
    vector<long long> ans(n);
    long long modulo = 998244353;
    for (int k=0; k<n; k++) {
        for (int i=0; i<n; i++) {
            C[i] = C[i] * (A[i] + B[k]) % modulo;
        }
        for (int i=0; i<n; i++) {
            ans[k] += C[i];
        }
        ans[k] %= modulo;
    }
    return ans;
}

int main() {
    long long n;
    cin >> n;
    vector<long long> A(n), B(n), C(n);
    for (int i=0; i<n; i++) cin >> A[i];
    for (int i=0; i<n; i++) cin >> B[i];
    for (int i=0; i<n; i++) cin >> C[i];
    vector<long long> ans = foo(n, A, B, C);
    for (int i=0; i<n; i++) {
        cout << ans[i];
        if (i<n-1) cout << ' ';
        else cout << endl;
    }
}

File: archived/2022-09/icpc-d.py
def foo(n, A, B, C):
    ls = C
    ans = [0] * n
    modulo = 998244353
    for k in range(n):
        for i in range(n):
            ls[i] = ls[i] * (A[i] + B[k]) % modulo
        ans[k] = sum(ls) % modulo
    return ans

n = int(input())
A = [int(i) for i in input().split()]
B = [int(i) for i in input().split()]
C = [int(i) for i in input().split()]
for i, val in enumerate(foo(n, A, B, C)):
    if i + 1 < n:
        print(val, end=' ')
    else:
        print(val)